modules/repository/generate.go

```diff
@@ -353,3 +371,13 @@ func GenerateRepository(ctx context.Context, doer, owner *user_model.User, templ
 
 	return generateRepo, nil
 }
+
+// Escapes user input to valid OS filenames
+//
+//		Based on https://github.com/sindresorhus/filename-reserved-regex
+//	 Adds "." to prevend directory traversal
+func fileNameEscape(s string) string {
```

Comment:
The name `fileNameEscape`: it doesn't do escaping (escaping means it can be un-escaped to the original value), but it it do sanitizing IMO.